pkg/skaffold/deploy/docker/deploy.go

```diff
@@ -0,0 +1,160 @@
+	monitor  status.Monitor
+	syncer   pkgsync.Syncer
+
+	cfg                *v1.DockerDeploy
+	client             dockerutil.LocalDaemon
+	deployedContainers map[string]string // imageName -> containerID
+	network            string
+	once               sync.Once
+}
+
+func NewDeployer(cfg dockerutil.Config, labeller *label.DefaultLabeller, d *v1.DockerDeploy, resources []*v1.PortForwardResource) (*Deployer, error) {
+	client, err := dockerutil.NewAPIClient(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Deployer{
+		cfg:                d,
+		client:             client,
+		deployedContainers: make(map[string]string),
+		network:            fmt.Sprintf("skaffold-network-%s", uuid.New().String()),
+		// TODO(nkubala): implement components
+		accessor: &access.NoopAccessor{},
+		debugger: &debug.NoopDebugger{},
+		logger:   &log.NoopLogger{},
+		monitor:  &status.NoopMonitor{},
+		syncer:   &pkgsync.NoopSyncer{},
+	}, nil
+}
+
+func (d *Deployer) TrackBuildArtifacts(artifacts []graph.Artifact) {
+	// TODO(nkubala): implement with components
+}
+
+func (d *Deployer) Deploy(ctx context.Context, out io.Writer, builds []graph.Artifact) error {
+	var err error
+	d.once.Do(func() {
+		err = d.client.NetworkCreate(ctx, d.network)
+	})
+	if err != nil {
+		return fmt.Errorf("creating skaffold network %s: %w", d.network, err)
+	}
+	for _, b := range builds {
+		// TODO(nkubala): parallelize this
+		if !util.StrSliceContains(d.cfg.Images, b.ImageName) {
+			continue
+		}
+		if containerID, found := d.deployedContainers[b.ImageName]; found {
+			logrus.Debugf("removing old container %s for image %s", containerID, b.ImageName)
+			if err := d.client.Delete(ctx, out, containerID); err != nil {
+				return fmt.Errorf("failed to remove old container %s for image %s: %w", containerID, b.ImageName, err)
+			}
+		}
+		if d.cfg.UseCompose {
+			// TODO(nkubala): implement
+			return fmt.Errorf("docker compose not yet supported by skaffold")
+		}
+		opts := dockerutil.ContainerCreateOpts{
+			Name:    b.ImageName,
+			Image:   b.Tag,
+			Network: d.network,
+		}
+		id, err := d.client.Run(ctx, out, opts)
+		if err != nil {
+			return errors.Wrap(err, "creating container in local docker")
+		}
+		d.deployedContainers[b.ImageName] = id
+	}
+
+	return nil
+}
+
+func (d *Deployer) Dependencies() ([]string, error) {
+	// noop since there is no deploy config
+	return nil, nil
+}
+
+func (d *Deployer) Cleanup(ctx context.Context, out io.Writer) error {
+	for _, id := range d.deployedContainers {
+		if err := d.client.Delete(ctx, out, id); err != nil {
+			// TODO(nkubala): replace with actionable error
+			return errors.Wrap(err, "cleaning up deployed container")
+		}
+	}
+
+	err := d.client.NetworkRemove(ctx, d.network)
+	return errors.Wrap(err, "cleaning up skaffold created network")
+}
+
+func (d *Deployer) Render(context.Context, io.Writer, []graph.Artifact, bool, string) error {
+	return errors.New("render not implemented for docker deployer")
```

Comment:
This could output a docker-compose file?

Should we define a fixed error instance for NOTSUPPORTED?